src/pages/iou/request/step/IOURequestStepAmount.tsx

```diff
@@ -136,6 +136,9 @@ function IOURequestStepAmount({
     };
 
     const navigateToCurrencySelectionPage = () => {
+        if (moneyRequestAmountInput?.current?.getCurrency() !== currency && moneyRequestAmountInput?.current?.getCurrency()) {
+            IOU.setMoneyRequestCurrency(transactionID, moneyRequestAmountInput?.current?.getCurrency(), action === CONST.IOU.ACTION.EDIT);
+        }
```

Comment:
This feels hacky. On navigation we should just navigate without doing side effect (changing currency). The currency should be changed when in fact we change it

Reply:
> The problem is when we open the currency page the selected currency is the currency param in the URL and when we choose the original currency, currencyFromProps isn't changed then currency prop can't be changed.

@s77rt As I mentioned above, the problem is here. Do you have any other ideas for this case?

Reply:
We shouldn't change the currency until we press the save button that will actually change the currency. Otherwise we will have bugs e.g.
1. Start money request flow with currency A and go to the edit amount page
2. Paste currency B
3. Press the currency button
4. Go back (cancel the edit process)

The currency should be still A